src/main/java/net/mcreator/plugin/events/ui/ModElementGUIEvent.java

```diff
@@ -19,21 +19,32 @@
 
 package net.mcreator.plugin.events.ui;
 
+import net.mcreator.ui.MCreator;
 import net.mcreator.ui.MCreatorTabs;
 import net.mcreator.ui.modgui.ModElementGUI;
 
-/**
- * <p>These events are triggered at different states of a {@link ModElementGUI} loading.</p>
- */
 public class ModElementGUIEvent extends TabEvent {
 
+	private final MCreator mcreator;
 	private final ModElementGUI<?> modElementGUI;
 
-	public ModElementGUIEvent(MCreatorTabs.Tab tab, ModElementGUI<?> modElementGUI) {
+	/**
+	 * <p>These events are triggered at different states of a {@link ModElementGUI} loading.
+	 * However, this constructor is never called outside this class.</p>
+	 * @param mcreator <p>The {@link MCreator} instance where the mod element is.</p>
+	 * @param tab <p>The {@link MCreatorTabs.Tab} that is opened for the {@link ModElementGUI}.</p>
+	 * @param modElementGUI <p>The {@link ModElementGUI} that calls each event.</p>
+	 */
+	public ModElementGUIEvent(MCreator mcreator, MCreatorTabs.Tab tab, ModElementGUI<?> modElementGUI) {

```

Comment:
Constructors that are never called outside this class should be protected not public, applies to all cases of this in events